### PR TITLE
[Enhancement] Print stack trace if cannot acquire lock for a long time

### PR DESCRIPTION
**Pull request by sduzh**
Print stack trace if cannot acquire lock for a long time. Used to
locate production environment deadlock issues.

Obtain stack trace through SQL:
```
> admin execute on 10006 '
for (s in ExecEnv.list_stack_trace_of_long_wait_mutex()) {
    System.print(s)
}
';
```

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by decster — COMMENTED** [review]
cache the trace to a global var, and make it available through a function like get_long_wait_mutex_info, bind it to the script engine

**Comment by sduzh** [comment]
> 

Ok, I'll update this later.

**Comment by sduzh** [comment]
> cache the trace to a global var, and make it available through a function like get_long_wait_mutex_info, bind it to the script engine

Updated, PTAL

**Comment by sduzh** [comment]
@imay @wyb PTAL

**Comment by imay** [comment]
How about adding some UT

**Comment by sduzh** [comment]
> How about adding some UT

OK

File: be/src/util/time.h [hunk 1]
```diff
@@ -34,6 +34,11 @@
 
 #pragma once
 
+#include <stdint.h>   // int64_t, uint64_t
```
[the remark this answers is not in the document]
Comment (sduzh) [reply be/src/util/time.h]: fixed

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**